src/components/CountryDetails.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+
+function CountryDetails(props) {
+
+const {alpha3Code} = useParams();
+const [country, setCountry] = useState(null);
+
+
+    setCountry(() => {
```

Comment:
here we dont need to use state. we have all the data we want in props . you can filter it directly and use it :

 const result = props.countries.filter((element) => {
    return element.alpha3Code === alpha3Code;
  });


  return (
    <>{result &&

      <h1>{result[0].name.official}</h1>
    }
    </>
  );

Reply:
you can also use find() method instead of filter() method<< find returns an object :)